publish.gradle

```diff
@@ -1,132 +1,222 @@
+    pubVersion = project.publishVersion
 } else {
-    ntVersion = WPILibVersion.version
+    pubVersion = WPILibVersion.version
 }
 
-if (project.hasProperty("utilPublishVersion")) {
-    utilVersion = project.utilPublishVersion
-} else {
-    utilVersion = "1.0.2${-> getVersion()}"
-}
+def outputsFolder = file("$buildDir/outputs")
 
-def utilFile = file("$buildDir/wpiutil.txt")
-def ntcoreFile = file("$buildDir/ntcore.txt")
+def versionFile = file("$outputsFolder/version.txt")
 
 task outputVersions() {
-    description = 'Prints the versions of ntcore and wpiutil to a file for use by the downstream packaging project'
+    description = 'Prints the versions of ntcore to a file for use by the downstream packaging project'
     group = 'Build'
-    outputs.files(utilFile, ntcoreFile)
+    outputs.files(versionFile)
 
     doFirst {
-        buildDir.mkdir()
+        outputsFolder.mkdir()
     }
 
     doLast {
-        utilFile.write utilVersion
-        ntcoreFile.write ntVersion
+        versionFile.write pubVersion
     }
 }
 
-task clean(type: Delete) {
-    delete utilFile
-    delete ntcoreFile
-}
+build.dependsOn outputVersions
 
-outputVersions.mustRunAfter clean
+def baseArtifactId = 'ntcore'
+def artifactGroupId = 'edu.wpi.first.ntcore'
 
-project(':native:wpiutil').build.dependsOn outputVersions
-project(':native:ntcore').build.dependsOn outputVersions
-if (project.buildArm) {
-    project(':arm:wpiutil').build.dependsOn outputVersions
-    project(':arm:ntcore').build.dependsOn outputVersions
-}
+def licenseFile = file("$rootDir/license.txt")
 
+task cppSourcesZip(type: Zip) {
+    destinationDir = outputsFolder
+    classifier = "sources"
 
+    from(licenseFile) {
+        into '/'
+    }
 
-// We change what repo we publish to depending on whether this is a development, beta, stable, or full
-// release. This is set up in the main gradle file.
-publishing {
-    publications {
-        def nat = project('native:ntcore')
-        if (!project.hasProperty('skipJava')) {
-            java(MavenPublication) {
-                artifact nat.jar
-                artifact nat.networktablesJavaSource
-                artifact nat.networktablesJavadoc
-                if (project.buildArm) {
-                    def natArm = project('arm:ntcore')
-                    artifact natArm.jar
-                    // If the library is not embedded include it in the repo
-                    if (!project.hasProperty('compilerPrefix')) {
-                        artifact natArm.ntcoreZip
-                    }
-                }
+    from('src/main/native/cpp') {
+        into '/'
+    }
 
-                if (project.hasProperty('makeDesktop')) {
-                    artifact nat.jar, {
-                        classifier = 'desktop'
+    model {
+        tasks {
+            it.each {
+                if (it.name.contains('jniHeaders')) {
+                    from (it.outputs.files) {
+                        into '/'
                     }
```

Comment:
Can you explain how that's possible? It seems like the receiver of the call to `from` is `project` not the task. As you said above, the task is of type `DefaultTask` which doesn't have a `from` method on it.

If the receiver of the `from` call is `project` then it would be executed immediately.

What is the receiver of the `from` call?